### PR TITLE
feat: create api login with email password

### PR DESCRIPTION
**Pull request by LongTCH**

### CODE REVIEW

**Comment by ces-hoangdao** [comment]
please resolve conflict

File: app/controllers/auth.js [hunk 1]
```diff
@@ -82,4 +82,30 @@ export const authController = {
       });
     }
   },
+
+  async loginWithEmailAndPassword(req, res) {
```
Comment (ces-hoangdao) [inline app/controllers/auth.js:86]: just loginWithEmail is enough

File: app/controllers/auth.js [hunk 2]
```diff
@@ -82,4 +82,30 @@ export const authController = {
...
+          message: RESPONSE_MESSAGE.EMAIL_LOGIN_ERROR,
+        });
+      }
+      delete user.password;
+      res.status(HTTP_STATUS.OK).json({
+        success: true,
+        data: user,
+      });
+    } catch {
+      res.status(HTTP_STATUS.INTERNAL_SERVER_ERROR).json({
```
Comment (ces-hoangdao) [inline app/controllers/auth.js:105]: we have util sendError right? please use it

File: app/controllers/auth.js (hunk 1 above)
Comment (LongTCH) [reply app/controllers/auth.js:86]: I did

File: app/controllers/auth.js (hunk 2 above)
Comment (LongTCH) [reply app/controllers/auth.js]: I did

File: app/services/auth.js [hunk 3]
```diff
@@ -23,4 +23,17 @@ export const authService = {
       .lean();
     return sendUser;
   },
+
+  async getLoggedInUserOfTypeEmail({ email, password }) {
+    let user = await authService.getUserOfTypeEmail(email);
```
Comment (ces-hoangdao) [inline app/services/auth.js:28]: const instead of let, you are not re-assigning it

Comment (LongTCH) [reply app/services/auth.js]: I fixed

File: app/controllers/auth.js [hunk 4]
```diff
@@ -82,4 +82,27 @@ export const authController = {
...
+        email,
+        password,
+      });
+      if (!user) {
+        return res.status(HTTP_STATUS.NOT_FOUND).json({
+          success: false,
+          message: RESPONSE_MESSAGE.EMAIL_LOGIN_ERROR,
+        });
+      }
+      delete user.password;
```
Comment (ces-hoangdao) [inline app/controllers/auth.js:99]: Should func getLoggedInUserOfTypeEmail you don't get password then you don't need this line

Comment (LongTCH) [reply app/controllers/auth.js]: I moved it to getLoggedInUserOfTypeEmail

**Review by ces-hoangdao — APPROVED** [review]

**Pull request merged**